backend/src/main/java/com/woowacourse/f12/application/review/ReviewEventListener.java

```diff
@@ -0,0 +1,21 @@
+package com.woowacourse.f12.application.review;
+
+import com.woowacourse.f12.application.product.ProductDeletedEvent;
+import com.woowacourse.f12.domain.review.ReviewRepository;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ReviewEventListener {
+
+    private final ReviewRepository reviewRepository;
+
+    public ReviewEventListener(final ReviewRepository reviewRepository) {
+        this.reviewRepository = reviewRepository;
+    }
+
+    @EventListener
+    public void handle(final ProductDeletedEvent event) {
+        reviewRepository.deleteByProductId(event.getProductId());
+    }
+}
```

Comment:
Here are some suggestions for improving the code:

1. Add comments to clarify the purpose of the class and method, as well as any important implementation details.
2. Consider adding a try-catch block to handle any exceptions that might occur when deleting reviews. For example, if a database connection error occurs, the review deletion could fail, and the exception should be logged and handled gracefully.
3. Consider making the method name more descriptive, such as "deleteReviewsForDeletedProduct", to make it easier to understand what the method does.
4. It's good that the ReviewRepository is being injected through the constructor. This is a good practice for dependency injection.
5. Consider adding unit tests to verify that the method works as expected.

Overall, the code looks good and follows good coding practices.

Reply:
LGTM